extensions/github-authentication/src/githubServer.ts

```diff
@@ -92,13 +92,13 @@ async function getUserInfo(token: string, serverUri: vscode.Uri, logger: Log): P
 		throw new Error(NETWORK_ERROR);
 	}
 
+	const json = await result.json();
```

Comment:
I am slightly concerned that the result might not always come back in the form of JSON... are you sure that the result will always be JSON?

Reply:
I think you are right, I also thought the same after I submitted the first version.

I added error handling to the JSON parsing. However I feel the code is a bit clumsy now. Do you have any recommendations regarding improving the code or do you think it is fine?